treeherder/model/migrations/0054_duplicate_failure_classifications.py

```diff
@@ -0,0 +1,172 @@
+# ALTER TABLE `text_log_error_match` ADD CONSTRAINT `text_log_error_match_matcher_id_9b9dbb12_fk_matcher_id` FOREIGN KEY (`matcher_id`) REFERENCES `matcher` (`id`);
+# ALTER TABLE `text_log_error` ADD CONSTRAINT `text_lo_best_classification_id_5a40639a_fk_classified_failure_id` FOREIGN KEY (`best_classification_id`) REFERENCES `classified_failure` (`id`);
+# ALTER TABLE `text_log_error` ADD CONSTRAINT `text_log_error_failure_line_id_c0d8c8b0_fk_failure_line_id` FOREIGN KEY (`failure_line_id`) REFERENCES `failure_line` (`id`);
+# ALTER TABLE `text_log_error_match` ADD CONSTRAINT `text_log_error_m_text_log_error_id_09109841_fk_text_log_error_id` FOREIGN KEY (`text_log_error_id`) REFERENCES `text_log_error` (`id`);
+# COMMIT;
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('model', '0053_add_job_platform_option_push_index'),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql="""
+-- Disable foreign key checks
+SET FOREIGN_KEY_CHECKS=0;
+
+-- Create model TextLogErrorMatch
+CREATE TABLE `text_log_error_match`
+    (`id` bigint AUTO_INCREMENT NOT NULL PRIMARY KEY,
+     `score` numeric(3, 2) NULL,
+     `classified_failure_id` bigint NOT NULL,
+     `matcher_id` integer NOT NULL,
+     `text_log_error_id` bigint NOT NULL,
+     INDEX `text_log_error_match_9f0d755f` (`classified_failure_id`),
+     INDEX `text_log_error_match_ad8a0085` (`text_log_error_id`),
+     CONSTRAINT `text_log_error_match_text_log_error_id_192a8c4c_uniq`
+         UNIQUE (`text_log_error_id`, `classified_failure_id`, `matcher_id`),
+     CONSTRAINT `text_log_classified_failure_id_1b38b499_fk_classified_failure_id`
+         FOREIGN KEY (`classified_failure_id`) REFERENCES `classified_failure` (`id`),
+     CONSTRAINT `text_log_error_match_matcher_id_9b9dbb12_fk_matcher_id`
+         FOREIGN KEY (`matcher_id`) REFERENCES `matcher` (`id`),
+     CONSTRAINT `text_log_error_m_text_log_error_id_09109841_fk_text_log_error_id`
+         FOREIGN KEY (`text_log_error_id`) REFERENCES `text_log_error` (`id`));
+
+-- Add field best_classification to textlogerror
+-- Add field best_is_verified to textlogerror
+-- Add field failure_line to textlogerror
+ALTER TABLE `text_log_error`
+    ADD COLUMN `best_classification_id` bigint NULL,
+    ADD COLUMN `best_is_verified` bool DEFAULT 0 NOT NULL,
+    ADD COLUMN `failure_line_id` bigint NULL UNIQUE,
+    ADD INDEX `text_log_error_7a340891` (`best_classification_id`),
+    ADD CONSTRAINT `text_lo_best_classification_id_5a40639a_fk_classified_failure_id`
+        FOREIGN KEY (`best_classification_id`) REFERENCES `classified_failure` (`id`),
+    ADD CONSTRAINT `text_log_error_failure_line_id_c0d8c8b0_fk_failure_line_id`
+        FOREIGN KEY (`failure_line_id`) REFERENCES `failure_line` (`id`);
+ALTER TABLE `text_log_error`
+    ALTER COLUMN `best_classification_id` DROP DEFAULT,
+    ALTER COLUMN `best_is_verified` DROP DEFAULT,
+    ALTER COLUMN `failure_line_id` DROP DEFAULT;
```

Comment:
Since the text_log_error table is currently append only (i.e. we don't change the data in any way), you can use the trick I outlined in my document of:

```sql
create table text_log_error_copy like text_log_error;
... make modifications to text_log_error_copy...
insert into text_log_error_copy select * from text_log_error;
SET @max_id = (select max(id) FROM text_log_error_copy);
insert into text_log_error_copy select * from text_log_error where id>@max_id; 
rename table text_log_error_copy to text_log_error, text_log_error to text_log_error_old;
```

(you may want to add in a lock table to do the last bit, should be easy but I haven't yet researched exactly how this would be done)